### DISCUSSION TITLE
Allow ignoring certain file extensions / patterns from watch rebuilding.

### DISCUSSION THREAD

**Discussion by tiagoad** [remix-run/remix#2693] [Proposals]
I'm running sass in watch mode in parallel with `remix watch` to build my scss files into a few main CSS files.

I want to keep my scss files next to my components, but unfortunately this means every time I change a style, remix rebuilds twice, increasing response time quite a bit (especially if I make many changes in a row).

It would be nice to have a way to ignore certain files (like .scss) from rebuilding the project. 

Even better would be a way to run the remix builder programmatically, and have my own watcher that calls remix build without all the startup overhead, as I want to fine tune debouncing further. (Is this currently possible?)

**Comment by tpaulshippy** [comment]
Similar request, but I would like to be able to ignore changes to test files.

**Comment by acusti** [comment]
i am also using SCSS and compiling to CSS during dev via `run-p` and the following run script: `"dev:sass": "sass --watch app/:app/",`

i have the same double-build problem, except that i have also observed that a significant portion of the time, the CSS changes aren’t actually picked up in the latest build and i have to trigger another save to get it to show up. it makes for a challenging dev experience when writing CSS (lots of “why didn’t my style changes work? did i mess up a selector, or maybe they’re being overridden, or i messed up the decls…? oh right, they didn’t actually get updated in the browser”). the existing configuration options are so close: [`watchPaths`](https://remix.run/docs/en/1.16.0/file-conventions/remix-config#watchpaths) is almost exactly what i want, except that it doesn’t allow me to override the use of `appDirectory` as the watch base with `app/**/*\.(?!scss)$` or similar.

**Comment by xendarboh** [comment]
What would be useful, is something like `watchIgnore` as a companion to `watchPaths` (with ability to ignore files within paths, as others explained). My context is running remix within a nx monorepo, and while watching for changes in other nx projects is ideal (as the remix app is using code from them), there are some aspects of those other projects that will completely spaz out remix's watch process. I would simply like to specify paths for remix watch to ignore and I did not find a way to do that. ~~A solution for me now, due to the conformity of my project libs, is to only watch for `src/` files.~~

```js
watchPaths: ['../../libs/*/src'],
```

Other projects within the nx monorepo are using watchman, and it works very well. My editor (neovim) is also using watchman, so besides the remix project(s) within the monorepo, there's a somewhat universal watching mechanism using a local `.watchmanconfig` with [ignore_dirs](https://facebook.github.io/watchman/docs/config.html#ignore_dirs) configuration.

I went to bump up this discussion and noticed I was here before... many months ago.  bump bump! :smile: 

With some digging, looks like remix is using [chokidar](https://github.com/paulmillr/chokidar) watching library:
https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler/watch.ts#L100

While unused by remix, chokidar has an `ignored` option that would suffice if `remix.config.js` settings were connected to it:
https://github.com/paulmillr/chokidar#path-filtering

**Comment by luisrudge** [comment]
using vite, you can use `server.watch.ignored` to achieve this:

```
  server: {
    port: 3000,
    watch: {
      ignored: ['**/tests/**', '*.test.ts'],
    },
  },
```

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
